File: backend/app/schemas/audit_log.py
from datetime import datetime
from pydantic import BaseModel


class AuditLogResponse(BaseModel):
    id: int
    user_email: str | None
    action: str
    resource_type: str
    resource_id: str | None
    status: str
    details: str | None
    execution_id: str | None
    created_at: datetime

    class Config:
        from_attributes = True

File: frontend/src/app/core/services/audit-log.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
// Use project root path for environments (resolves module lookup issues)
import { environment } from 'environments/environment';

export interface AuditLog {
  id: number;
  user_email?: string;
  action: string;
  resource_type: string;
  resource_id?: string;
  status: string;
  details?: string;
  execution_id?: string;
  created_at: string;
}

@Injectable({ providedIn: 'root' })
export class AuditLogService {
  constructor(private http: HttpClient) {}

  listAuditLogs() {
    return this.http.get<AuditLog[]>(`${environment.apiUrl}/audit-logs`);
  }
}